_posts/2019-08-01-dropping-support-for-mysql.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: Dropping support for MySQL
+date: 2019-08-01 9:34:21
+author: Lukas Zapletal
+tags:
+- foreman
+---
+
+Foreman is essentially a Ruby on Rails application. A big one. And Ruby on
+Rails applications usually support multiple relational databases thanks to
+ActiveRecord stack. That's also the case for Foreman. But it's not that easy.
+
+<!--more-->
+
+Foreman currently requires one of the following databases:
+
+* PostgreSQL 9.2 or newer
+* MySQL 5.5 or newer
+
+This is not an extensive list you might think. Supporting two databases must
+not be that hard. Truth is, it's a challenge.
+
+Those two items on the list are both relational database systems with SQL
+standard, the problem is that very often behavior is different. Most of our
+development team use PostgreSQL and there are dozens of things that are
+different in MySQL. Therefore we need to identify those problems, find a way to
+fix them and then carry extra bit of code in our codebase for life. This takes
+a lot of our time we could invest in features, bugfixing or performance
+improvements.
+
+We have dicussed preferring PostgreSQL many times, but [the last time was
+different](https://community.theforeman.org/t/minimum-rdbm-requirements/14127)
+and we came to conclusion that dropping MySQL the right thing to do. [The
+plan](https://community.theforeman.org/t/rfc-dropping-support-for-mysql/14185)
+was then set and we are at the beginning of the journey where the first and
+really difficult step is on you, our users:
+
+**Foreman 1.24 will be the last version with MySQL support.**
+
+The upcoming versions 1.23 and 1.24 will also send a notification to all
+administrators in case it's running on MySQL. Don't worry, we will let you
+close the warning as soon as you see it, but we want to communicate this
+properly and in advance. We highly recommend to do the database migration to
+PostgreSQL during upgrade to version 1.23 because if there is a blocker bug
+that prevents from migrating to MySQL, there is still a backup plan of holding
+off migration and staying on MySQL for one more release until we come up with
+versions 1.24 or 1.25 which will be PostgreSQL only. This is important:
+
+**Do not wait with the migration for Foreman 1.24.**
+
+Along with the 1.23 release, we will provide official [migration
+documentation](/manuals/nightly/index.html#3.7MigratingtoPostgreSQL), and
```

Comment:
```suggestion
documentation](/manuals/1.23/index.html#3.7MigratingtoPostgreSQL), and
```

Reply:
Why not to point to the most recent version when a random user finds the post after some time? Updated instructions might be quite helpful I am pretty sure we will run into issues.

Reply:
because we will not have that section in the manual once we drop mysql

Reply:
Why would we ever remove section named "Migration from MySQL to PostgreSQL" in the future? People are far behind with upgrading, we should keep it.

Reply:
because we don't want to carry around stuff in the manual that isn't used anymore, if people in the future do the upgrade they will still find the instructions in the 1.23/1.24 manual